src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -224,6 +224,7 @@ internal IOperation CreateInternal(BoundNode boundNode)
                 case BoundKind.LocalDeclaration:
                     return CreateBoundLocalDeclarationOperation((BoundLocalDeclaration)boundNode);
                 case BoundKind.MultipleLocalDeclarations:
+                case BoundKind.UsingLocalDeclarations:
```

Comment:
>case BoundKind.UsingLocalDeclarations: [](start = 16, length = 38)

It looks like changes made to bound trees to support Using Declarations violate the invariant that only leaf nodes can have kinds. I.e. base nodes shouldn't have kinds. This way we can be sure that type checks for nodes that have kinds are equivalent to kind checks.

Reply:
So do you want me to change this as part of this PR, or should we just log a bug and fix it in the future?

Reply:
> So do you want me to change this as part of this PR, or should we just log a bug and fix it in the future?

I think it is a simple fix - introduction of a common base node. Should fit into this PR.

---
In reply to: [333678166](https://github.com/dotnet/roslyn/pull/39125#discussion_r333678166) [](ancestors = 333678166)

Reply:
Done.